Go to comments screen upon 1st comment addition

miankhalid opened the pull request:
Fixes [MA-2191](https://openedx.atlassian.net/browse/MA-2191)

Comment posted notification removed from the responses screen, since we are now navigating to comments screen every time a comment is posted.

@1zaman @BenjiLee plz review

1zaman commented on VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionCommentsFragment.java:
```diff
@@ -111,6 +114,10 @@ public void onClick(View v) {
                     }
                 });
+
+        if (isNewCommentPosted) {
```
This flag should be set as `false` after the first use, as `onViewCreated()` may be called multiple times.

1zaman commented on VideoLocker/src/main/java/org/edx/mobile/view/DiscussionAddCommentFragment.java:
```diff
@@ -137,9 +137,14 @@ private void createComment() {
-                EventBus.getDefault().post(new DiscussionCommentPostedEvent(thread, discussionResponse));
+            public void onSuccess(@NonNull DiscussionComment comment) {
+                logger.debug(comment.toString());
+                if (discussionResponse.getChildCount() == 0) {
```
Perhaps add a comment explaining the desired flow here.

1zaman commented on VideoLocker/src/main/java/org/edx/mobile/view/Router.java:
```diff
@@ -212,10 +213,11 @@ public void showCourseDiscussionAddPost(@NonNull Activity activity, @Nullable Di
     }
 
-    public void showCourseDiscussionComments(Context context, DiscussionComment comment, DiscussionThread discussionThread) {
+    public void showCourseDiscussionComments(@NonNull Context context, @NonNull DiscussionComment comment, @NonNull DiscussionThread discussionThread, boolean isNewCommentPosted) {
```
Maybe add a new default method definition that calls to this one, with the last parameter provided as `false` by default.

1zaman commented:
Other than a few small issues and nitpicks 👍

miankhalid commented:
I am holding up on addressing the review changes, as I found out that there is an open discussion on whether we want this change or not [here](https://openedx.atlassian.net/browse/MA-2193) (currently waiting @BenjiLee's feedback).

BenjiLee commented:
Commented on https://openedx.atlassian.net/browse/MA-2193, also waiting on that discussion before reviewing this PR.

miankhalid commented:
Closing this PR.

Reason [(link)](https://openedx.atlassian.net/browse/MA-2193?focusedCommentId=183797&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-183797):
>As long as the previous view we go back to is at the same state (scrolled to the same place) than no change is needed.
For example, if the response is near the bottom of the page and I add the first comment, after adding that first comment, I will be taken back to the bottom of that page where I found the response.
